File: dehaze-python/app/models/entity/sys_ai_model_price.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import BigInteger, DateTime, Integer, Numeric, SmallInteger, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysAiModelPrice(BaseModel, SoftDeleteMixin):
    """AI模型用户售价版本主表（价格版本化，单价以档位明细行表达）"""

    __tablename__ = "sys_ai_model_price"
    __table_args__ = {"comment": "AI模型用户售价版本表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    model_id: Mapped[str] = mapped_column(
        String(64), nullable=False, comment="模型标识(关联sys_ai_model.model_id)"
    )
    provider_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment="供应商ID")
    price_version: Mapped[int] = mapped_column(
        Integer, nullable=False, default=1, comment="价格版本号(同模型同供应商内递增)"
    )
    unit: Mapped[str] = mapped_column(
        String(24), nullable=False, default="credits_per_million",
        comment="单价单位(credits_per_million:积分/百万token)",
    )
    effective_from: Mapped[datetime] = mapped_column(DateTime, nullable=False, comment="价格版本生效时间")
    effective_to: Mapped[datetime | None] = mapped_column(
        DateTime, nullable=True, comment="价格版本失效时间(NULL表示当前版本)"
    )
    status: Mapped[int] = mapped_column(
        SmallInteger, nullable=False, default=1, comment="状态(1:生效;0:停用)"
    )


class SysAiModelPriceDetail(BaseModel, SoftDeleteMixin):
    """AI模型用户售价档位明细表（token类型 × 上下文分段 × 时段）"""

    __tablename__ = "sys_ai_model_price_detail"
    __table_args__ = {"comment": "AI模型用户售价档位明细表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    price_id: Mapped[int] = mapped_column(
        BigInteger, nullable=False, index=True, comment="价格版本ID(关联sys_ai_model_price.id)"
    )
    token_type: Mapped[str] = mapped_column(
        String(16), nullable=False, comment="token类型(input;cached;output)"
    )
    time_slot: Mapped[str] = mapped_column(
        String(16), nullable=False, comment="时段档位(peak;idle)"
    )
    min_tokens: Mapped[int] = mapped_column(
        BigInteger, nullable=False, default=0, comment="上下文分段下界"
    )
    max_tokens: Mapped[int | None] = mapped_column(
        BigInteger, nullable=True, comment="上下文分段上界(NULL不限)"
    )
    unit_price: Mapped[Decimal] = mapped_column(
        Numeric(12, 4), nullable=False, default=0, comment="单价(积分/百万token)"
    )

File: dehaze-go/internal/model/sys_ai_model_price.go
package model

import "time"

// SysAiModelPrice 模型用户售价版本主表
type SysAiModelPrice struct {
	BaseModel
	ModelID       string     `gorm:"column:model_id;type:varchar(64);not null;comment:模型标识" json:"modelId"`
	ProviderID    int64      `gorm:"column:provider_id;type:bigint;not null;index:idx_provider;comment:供应商ID" json:"providerId"`
	PriceVersion  int        `gorm:"column:price_version;type:int;not null;default:1;comment:价格版本号" json:"priceVersion"`
	Unit          string     `gorm:"column:unit;type:varchar(24);not null;default:credits_per_million;comment:单价单位" json:"unit"`
	EffectiveFrom time.Time  `gorm:"column:effective_from;type:datetime;not null;comment:生效时间" json:"effectiveFrom"`
	EffectiveTo   *time.Time `gorm:"column:effective_to;type:datetime;comment:失效时间(NULL表示当前版本)" json:"effectiveTo"`
	Status        int8       `gorm:"column:status;type:tinyint;not null;default:1;comment:状态(1:生效;0:停用)" json:"status"`
	Deleted       int64      `gorm:"column:deleted;type:bigint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysAiModelPrice) TableName() string {
	return "sys_ai_model_price"
}

// SysAiModelPriceDetail 用户售价档位明细（token类型 × 上下文分段 × 时段档位）
type SysAiModelPriceDetail struct {
	BaseModel
	PriceID   int64   `gorm:"column:price_id;type:bigint;not null;index:idx_price;comment:价格版本ID" json:"priceId"`
	TokenType string  `gorm:"column:token_type;type:varchar(16);not null;comment:计费类型(input/cached/output)" json:"tokenType"`
	TimeSlot  string  `gorm:"column:time_slot;type:varchar(16);not null;comment:时段档位(peak/idle)" json:"timeSlot"`
	MinTokens int64   `gorm:"column:min_tokens;type:bigint;not null;default:0;comment:上下文分段下界" json:"minTokens"`
	MaxTokens *int64  `gorm:"column:max_tokens;type:bigint;comment:上下文分段上界" json:"maxTokens"`
	UnitPrice float64 `gorm:"column:unit_price;type:decimal(12,4);not null;default:0;comment:单价(积分/百万token)" json:"unitPrice"`
	Deleted   int64   `gorm:"column:deleted;type:bigint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysAiModelPriceDetail) TableName() string {
	return "sys_ai_model_price_detail"
}
